addons/rating/models/rating.py

```diff
@@ -18,106 +18,156 @@ def _compute_res_name(self):
+            if rating:
+                values['rating_id'] = rating.id
+                rating.reset()
+            token = Token.create(values)
+            Token |= token
+        return token
+
+    def _rating_get_partner_id(self):
+        if hasattr(self, 'partner_id') and self.partner_id:
+            return self.partner_id
+        return self.env['res.partner']
+
+    def _rating_get_rated_partner_id(self):
+        if hasattr(self, 'user_id') and self.user_id.partner_id:
+            return self.user_id.partner_id
+        return self.env['res.partner']
+
+    @api.multi
+    def rating_send_request(self, template, partner_id=None, rated_partner_id=None, reuse_rating=True):
+        """ This method send rating request by email, using a template given
+        in parameter. """
+        if partner_id is None:
+            partner_id = self._rating_get_partner_id()
+        if rated_partner_id is None:
+            rated_partner_id = self._rating_get_rated_partner_id()
+        tokens = self.rating_get_request(partner_id, rated_partner_id, reuse_rating=reuse_rating)
+        for token in tokens:
+            template.send_mail(token.id, force_send=True)
+
+    @api.multi
+    def rating_apply(self, rate, token=None):
+        """ Apply a rating given a token. If the current model inherits from
+        mail.thread mixing, a message is posted on its chatter.
+
+        :param rate : the rating value to apply
+        :type rate : float
+        :param token : access token
+        :returns rating.rating record
+        """
+        Rating, rating = self.env['rating.rating'], None
+        if token:
+            token_rec = self.env['rating.token'].search([('access_token', '=', token)], limit=1)
+            if token_rec.rating_id:
+                rating = token_rec.rating_id
             else:
-                rating = Rating.create(values)
-            # send the mail
-            template.send_mail(rating.id, force_send=True)
+                rating = Rating.create({
+                        'res_model': token_rec.res_model,
+                        'res_id': token_rec.res_id,
+                        'partner_id': token_rec.partner_id.id,
+                        'rated_partner_id': token_rec.rated_partner_id.id
+                    })
+                token_rec.rating_id = rating
+        else:
+            rating = Rating.search([('res_model', '=', self._name), ('res_id', '=', self.ids[0])], limit=1)
+        if rating:
+            rating.write({'rating': rate})
+            if hasattr(self, 'message_post'):
+                self.message_post(
+                    body="%s %s <br/><img src='/rating/static/src/img/rating_%s.png' style='width:20px;height:20px'/>"
+                    % (rating.sudo().partner_id.name, _('rated it'), rate),
+                    subtype='mail.mt_comment',
+                    author_id=rating.partner_id and rating.partner_id.id or None  # None will set the default author in mail_thread.py
```

Comment:
partner_id is recordset. return False if empty recordset. no need to set None.
author_id=rating.partner_id.id